CollAction/Services/Donation/DonationService.cs

```diff
@@ -0,0 +1,165 @@
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
+using Stripe;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CollAction.Services.Donation
+{
+    public class DonationService : IDonationService
+    {
+        private readonly CustomerService _customerService;
+        private readonly SourceService _sourceService;
+        private readonly ApplicationDbContext _context;
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly RequestOptions _requestOptions;
+        private readonly SiteOptions _siteOptions;
+
+        public DonationService(IOptions<RequestOptions> requestOptions, IOptions<SiteOptions> siteOptions, UserManager<ApplicationUser> userManager, ApplicationDbContext context)
+        {
+            _requestOptions = requestOptions.Value;
+            _siteOptions = siteOptions.Value;
+            _customerService = new CustomerService(_requestOptions.ApiKey);
+            _sourceService = new SourceService(_requestOptions.ApiKey);
+            _context = context;
+            _userManager = userManager;
+        }
+
+        public async Task<string> InitializeCreditCardCheckout(string currency, int amount, string name, string email)
+        {
+            ValidateDetails(amount, name, email);
+
+            ApplicationUser user = await _userManager.FindByEmailAsync(email);
+            Customer customer = await GetOrCreateCustomer(name, email);
+
+            using (var client = new HttpClient())
+            {
+                var requestContent = new Dictionary<string, string>()
+                {
+                    { "success_url", $"{_siteOptions.PublicAddress}/Donation/ThankYou" },
+                    { "cancel_url", $"{_siteOptions.PublicAddress}/Donation/Donate" },
+                    { "payment_method_types[]", "card" },
+                    { "line_items[][amount]", (amount * 100).ToString() },
+                    { "line_items[][currency]", currency },
+                    { "line_items[][name]", "donation" },
+                    { "line_items[][description]", "A donation to Stichting CollAction" },
+                    { "line_items[][quantity]", "1" },
+                    { "customer", customer.Id }
+                };
+
+                using (var request = new HttpRequestMessage()
+                {
+                    RequestUri = new Uri("https://api.stripe.com/v1/checkout/sessions"),
+                    Method = HttpMethod.Post,
+                    Content = new FormUrlEncodedContent(requestContent)
+                })
+                {
+                    request.Headers.Authorization = new AuthenticationHeaderValue("Basic", Convert.ToBase64String(Encoding.UTF8.GetBytes(_requestOptions.ApiKey + ":")));
+                    request.Headers.Add("Stripe-Version", "2019-03-14; checkout_sessions_beta=v1");
```

Comment:
Does stripe require "date of API version" to process requests ? Could extract it to app--settings in future so that we can quickly change it following them, but I think it's not critical.

Reply:
We're using a beta stripe API here, because other flows don't comply with the upcoming SCA regulation (https://stripe.com/docs/strong-customer-authentication). That's why I'm doing this one in http, and the other one through the .net library. Otherwise we'd have to revise our entire page in half a year, now we only need to upgrade to the stable api version once it's ready.